index.js

```diff
@@ -3,14 +3,18 @@ var recast = require('recast');
 
 var transformer = require('./unreachableBranchTransformer');
 
-module.exports = function (file, opts) {
-  opts = opts || [];
-  var ignore = ['.json'].concat(opts.ignore || []).some(function(ext) {
-    return file.indexOf(ext, file.length - ext.length) !== -1;
-  });
+module.exports = function () {
+  var file = typeof arguments[0] == 'string' ? arguments[0] : null;
+  var opts = (typeof arguments[0] == 'string' ? arguments[1] : arguments[0]) || {};
+
+  if (file){
```

Comment:
I hear you, but I actually found your version harder to reason about. Agree to disagree :smile:

Since it seems that `if (typeof file === 'string') {` solves your problem and doesn't impact existing functionality, I'd more than happy to merge and publish that change.